app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def wrapper(*args: Any) -> Any:
+        cache_key = args
+        if cache_key in cache_storage:
+            print("Getting from cache")
+            return cache_storage[cache_key]
+        else:
+            print("Calculating new result")
```

Comment:
The print statements are useful for debugging but should be removed or replaced with logging before committing the code, as per the 'Clean Code' checklist item.